anesthetic/_code_utils.py

```diff
@@ -0,0 +1,53 @@
+def replace_inner_function(outer, new_inner):
+    """Replace a nested function code object used by outer with new_inner.
+
+    The replacement new_inner must use the same name and must at most use the
+    same closures as the original.
+
+    Code taken from https://stackoverflow.com/a/27550237
```

Comment:
This is a very fancy bit of python taken from the above link. It allows one to monkey patch a function within a function. This is only necessary for modifying the `pandas.plotting._matplotlib.boxplot.boxplot` command, which is infuriatingly written as a function rather than a class.